src/services/movieService.ts

```diff
@@ -0,0 +1,69 @@
+import { MovieRepository } from '../repositories/movieRepository';
+import { MovieTrailerDetail, ParseMovieData } from '../dto/movie.dto';
+import { YoutubeApi } from '../utils/youtube';
+import { MovieLike } from '../dto/movie.dto';
+import dotenv from 'dotenv';
+import MyCustomError from '../utils/customError';
+
+dotenv.config();
+
+class MovieService {
+  constructor(private Repository: MovieRepository, private Youtube: YoutubeApi) {}
+
+  public getMovieTrailerDetail = async (movieId: number, skip: number, take: number, userId: number): Promise<ParseMovieData> => {
+    const movieTrailerDetailData: MovieTrailerDetail = await this.Repository.getMovieTrailerDetail(movieId, userId);
+    const categoryId = movieTrailerDetailData?.category?.id;
+    const movieDetailAndMore = await this.Repository.getMovieDetailAndMore(movieId, skip, take, Number(categoryId));
+
+    const mainVideo = await this.Youtube.getMovieTrailerVideoId(movieTrailerDetailData.name);
+    const playlistVideoTitle = await this.Youtube.getMovieTrailerPlaylist(movieTrailerDetailData.name);
```

Comment:
```suggestion
    const [mainVideo, playlistVideoTitle] = await Promise.all([
      this.Youtube.getMovieTrailerVideoId(movieTrailerDetailData.name),
      this.Youtube.getMovieTrailerPlaylist(movieTrailerDetailData.name)
    ]);
```
이렇게 병렬처리 하는방법은 어떨까요?
테스트 해보고 문제가 없다면 제 생각에는 이게 더 좋아보입니다.

Reply:
수정 온료!